Add devcontainer

yubiuser opened the pull request:
Adds a devcontainer.

DL6ER reviewed (changes requested):
Inside the container, note the last line:
```
/workspaces/docs $ npm test

> docs@1.0.0 test
> npm run markdownlint && npm run linkinator


> docs@1.0.0 markdownlint
> markdownlint-cli2 "**/*.md" "!**/node_modules/**"

markdownlint-cli2 v0.12.1 (markdownlint v0.33.0)
Finding: **/*.md !**/node_modules/**
Linting: 84 file(s)
Summary: 0 error(s)

> docs@1.0.0 linkinator
> npm run build && linkinator site --recurse --silent --skip "^(?!http://localhost)"


> docs@1.0.0 build
> mkdocs build --clean

sh: mkdocs: not found
```
---

On other notes: We should also add a `serve` command (and port-forward) so you can browse the docs while editing it in your devcontainer. I am running
```
mkdocs serve --dev-addr 0.0.0.0:8000
```
to achieve this when running bare-metal.

yubiuser commented:
The issue should be fixed now. Reason it didn't work was because

```
export PATH="$HOME/.local/bin:$PATH"
```
was executed as `postCreateCommand` but only valid in the creating shell, but not afterwards.

____

I also added `serve` command. `vscode` has an auto-forward port detection, but I added port 8000 anyways in the `devcontainer.json`

PromoFaux reviewed (approved):
LGTM

![image](https://github.com/pi-hole/docs/assets/1998970/bc1dd35f-0483-4190-ac69-2a050a316424)

edit: Also tested with `npm run serve` (and `npm run test`)
